lib/identity_cache/query_api.rb

```diff
@@ -113,6 +113,12 @@ def instrumented_record_from_coder(coder) #:nodoc:
         end
       end
 
+      def expire_primary_index(id) # :nodoc:
```

Comment:
If we want people to use it we should remove the `nodoc` and also document what it does, probably with a better name. This name is confusing. For example, when I see this method call I think: expire the primary index of what?

Reply:
I've updated the public method to be named `expire_primary_cache_index_by_id` and added a description.

Reply:
> For example, when I see this method call I think: expire the primary index of what?

We have multiple references to this primary index, e.g.
* IdentityCache::WithoutPrimaryIndex
* primary_cache_index_enabled
* https://github.com/Shopify/identity_cache#basic-usage briefly explains that fetching by id is using the primary index

Perhaps it would be clearer if we referred to it as the primary **key** cache index for consistency with `primary_key` active record method and the concept of a primary key in the database.

> I've updated the public method to be named `expire_primary_cache_index_by_id`

Since the primary cache index is always by id, the `by_id` suffix seems kind of redundant and like there could be some primary cache index by something else.  Instead, I think `expire_primary_key_cache_index` would be a better name for this method by just clarifying the reference to the primary key.